File: test/fixtures/php-guzzle/post_basic_auth_url_encoded_data.php
<?php
require 'vendor/autoload.php';

use GuzzleHttp\Client;

$client = new Client();

$response = $client->post('http://localhost:28139/api/oauth/token/', [
    'headers' => [
        'Content-Type' => 'application/x-www-form-urlencoded'
    ],
    'auth' => ['foo', 'bar'],
    'form_params' => [
        'grant_type' => 'client_credentials'
    ]
]);

File: test/fixtures/java-unirest/post_basic_auth_url_encoded_data.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.post("http://localhost:28139/api/oauth/token/")
            .basicAuth("foo", "bar")
            .header("Content-Type", "application/x-www-form-urlencoded")
            .field("grant_type","client_credentials")
            .asString();
    }
}
